tests/README.md

```diff
@@ -1,25 +1,239 @@
+- [Legacy Stuff (TODO -- Remove)](#legacy-stuff-todo----remove)
+    - [-](#--1)
+    - [OS X](#os-x)
+    - [Known issues](#known-issues)
+    - [Tips and Tricks](#tips-and-tricks)
+
+<!-- markdown-toc end -->
+
+
+-------------------------------------------------------------------------------
+
+
+## Getting Started
+
+The dependencies for the integration tests are collected and organized in
+dependency files in the `./requirements` folder, and separated into
+
+
+> These requirement files are using the *hashicorp/bionic64* *Vagrant* image is
+> as the basis for the requirements files generated.
+
+| Debian | Alpine | Python |
+| ------------- | :-------------: | -----: |
+| *deb-requirements.txt* | *apt-requirements.txt* | *python-requirements.txt* |
+
+## Installing Dependencies
+
+### Debian
+
+```bash
+$ apt install -yyq $(cat requirements/integration/deb-requirements.txt)
+```
+
+### Alpine Linux
+
+```bash
+# TODO -- What is the correct command for the Alpine package manager
+$ apk add -yyq $(cat requirements/integration/apk-requirements.txt)
+```
+
+### Python2 and Python3
+
+```bash
+$ pip install -r requirements/requirements/python-requirements.txt
+$ pip3 install -r requirements/requirements/python-requirements.txt
+```
+
+
+-------------------------------------------------------------------------------
+
+
+## Running the Tests Locally
+
+> The tests can be run locally without any further involvement as long as all
+> the dependencies have been installed and are at the correct version. However,
+> managing dependencies, especially with Python can be a hassle. Therefore it is
+> recommended to add a virtual Python environment to isolate the dependencies
+> needed for running the integration tests.
+
+### Isolating Python Dependencies
+
+In order to avoid dependency mismanagement due to Python packages differing from
+one test environment to the other, it is recommended to use a python virtual
+environment. The "de-facto" standard is
+[virtualenv](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/).
+
+#### Install
+
+```bash
+$ pip install virtualenv
+```
+
+#### Initialize the Virtual Environment
+
+```bash
+$ cd integration-dir/tests
+$ virtualenv <name-of-virtualenv-folder>
+```
+
+##### (Optional) -- Select Python Version
+```bash
+$ virtualenv -p /usr/bin/python2.7 <name-of-virtualenv-folder>
+```
+
+#### Activate the Virtual Environment
+
+```bash
+$ source <name-of-virtualenv-folder>/bin/activate
+```
+
+This now means that you have a clean python environment, and no packages you
+have previously installed outside of this virtual environment will be
+discoverable by Python.
+
+#### Install the Python Dependencies from the Requirements File
+
+```bash
+$ pip install -r requirements/integration/python-requirements.txt
+```
+
+This will install the requirements needed to run the tests.
+
+### Running the Tests
+
+Next, run the tests with the `run.sh` script.
+
+```bash
+$ ./run.sh
```

Comment:
We can fix independently, but there is at least one dependency missing in `run.sh`: the update module generators. To fix this we should donwload them there.

Reply:
Mmm. Maybe I am wrong. Have you managed to run all the tests with just the dependencies installed from the run.sh wrapper?

Reply:
Just locally, and I might have all the dependencies already installed. I'll do a full run on the VM to verify.

Reply:
Has this been verified, then?